bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -41,6 +42,22 @@ export class AxisView extends GuideRendererView {
   panel: Panel
   layout: Layoutable
 
+  axis_label_math_text_view: MathTextView
+
+  /**
+   * Lazy initialize is called when views are instantiated,
+   * working like an async constructor
+   */
+  override async lazy_initialize(): Promise<void> {
+    await super.lazy_initialize()
+
+    const {axis_label} = this.model
+
+    // Build math_text_view if axis_label is a MathText instance
+    if (axis_label != null && axis_label instanceof MathText)
+      this.axis_label_math_text_view = await build_view(axis_label, {parent: this})
+  }
```

Comment:
I would say this special casing is the biggest deficiency of the current PR. The axis (and also: labels, titles, etc... anything else) should not have to explicitly care whether it is rendering plain text, or math text. It just has an object representing one or the other, and then asks that object to render itself in a particular place. With this approach everything that wants to render math text will not be insulated from the details it should be.

Ideally I would want to try to sort this up front but given the time-limited nature of this work I think we will have to merge this soon as a first-cut and sort out the design iteratively in future PRs. Again, @mattpap what is missing here is your input on the design you would want to see, how that intersects with your broader plans, and some estimates of timing. If you don't have the bandwidth to provide design guidance, then that is what it is, but we will need to move forward with something, regardless.

Reply:
This could, but I'm not 100% sure, be resolved with the use of the worker approach. I could just add the worker call inside TextBox but there would still need to be somewhat of a flag to tell render API it wants the text converted.